chia/wallet/cat_wallet/cat_wallet.py

```diff
@@ -783,6 +783,7 @@ async def generate_signed_transaction(
     ) -> List[TransactionRecord]:
         # (extra_delta, tail_reveal, tail_solution)
         cat_discrepancy: Optional[Tuple[int, Program, Program]] = kwargs.get("cat_discrepancy", None)
+        override_memos: Optional[bool] = kwargs.get("override_memos", None)
```

Comment:
I just want to make sure I have this straight in my head:
* You use this override_memos argument when you are sending a CAT to a DAO
* The format of this CAT is CAT(inner_puzzle)
* The DAO is recieving this CAT at CAT(p2_singleton_of_dao)

Is that right?  Because it seems to me that the DAO wallet could subscribe to its p2_singleton address and then that would be what the CAT normally hints to as well right?

Reply:
Yes that's right...the trouble is that p2_singleton_of_dao is unique for each CAT's asset id so everyone's DAO wallet would have to subscribe to all the possible p2_singleton addresses for every CAT in existence.

If you're the one sending the funds, you can subscribe to the p2_singleton puzhash, but then nobody else's DAO wallet would know to do that. The only way we have (currently) to make sure everyone can see incoming funds is to use the dao ID as the hint. The only alternative I can see would be for people to manually create a list of CAT's and have their wallet subscribe to just those assets.

Reply:
I'm not quite sure I'm following.  p2_singleton is parameterized by only the DAO's singleton ID.  Sure each cat will have a different FULL puzzle but the inner puzzle will be the same across all.  It's actually less permutations that our existing system because the standard puzzle can have derivations but p2_singleton is always p2_singleton.

Reply:
yeah this makes sense. I think in an earlier the version the asset ID was being used somewhere that made this not possible, but you're absolutely right and your suggested way works better. Have removed the cat wallet override, and adapted the dao chialisp, drivers and wallet to hint on the p2_singleton. Thanks for your persistence :)